recipe/conda_build_config.yaml

```diff
@@ -1,3 +1,5 @@
+c_compiler_version:        # [osx and arm64]
+  - 17                     # [osx and arm64]
```

Comment:
The change from 17 to 18 was the only obvious one between when we last built successfully and now, when no osx-arm64 builds seem to be working.

Reply:
This didn't help, so I have force-pushed to remove it.